Artifact/Model/ArtifactScenesViewModel.swift

```diff
@@ -0,0 +1,128 @@
+import Foundation
+import RealityKit
+import ArtifactScenes
+
+enum SceneLoadingState {
+    case notLoaded
+    case loading
+    case loaded
+    case failed(Error)
+}
+
+class ArtifactScenesViewModel: ObservableObject {
```

Comment:
Consider using the Observation framework instead of ObservableObjects. Your app target is 18.1, so there is no reason to use the old API.